lib/matobj2.gd

```diff
@@ -801,520 +1021,932 @@ DeclareOperation( "MutableCopyMat", [IsMatrixObj] );
 ##  <ManSection>
 ##  <Oper Name="CopySubMatrix" Arg='src, dst, srows, drows, scols, dcols'/>
 ##
-##  <Description>
 ##  <Returns>nothing</Returns>
-##  Does <C><A>dst</A>{<A>drows</A>}{<A>dcols</A>} := 
+##
+##  <Description>
+##  Does <C><A>dst</A>{<A>drows</A>}{<A>dcols</A>} :=
 ##  <A>src</A>{<A>srows</A>}{<A>scols</A>}</C>
-##  without creating an intermediate object and thus - at least in
-##  special cases - much more efficiently. For certain objects like
-##  compressed vectors this might be significantly more efficient if 
-##  <A>scols</A> and <A>dcols</A> are ranges with increment 1.
+##  without creating an intermediate object and thus
+##  &ndash;at least in special cases&ndash;
+##  much more efficiently.
+##  For certain objects like compressed vectors this might be
+##  significantly more efficient if <A>scols</A> and <A>dcols</A> are
+##  ranges with increment 1.
 ##  </Description>
 ##  </ManSection>
 ##  <#/GAPDoc>
 ##
-DeclareOperation( "CopySubMatrix", [IsMatrixObj,IsMatrixObj,
-                                    IsList,IsList,IsList,IsList] );
+DeclareOperation( "CopySubMatrix",
+    [ IsMatrixObj, IsMatrixObj, IsList, IsList, IsList, IsList ] );
 
-# TODO: order of arguments? dst before src??? perhaps:
-#    dst, drows, dcols,  src, srows, scols
-
-############################################################################
-# New element access for matrices
-############################################################################
 
 #############################################################################
 ##
-#o  <mat>[<row>,<col>] . . . . . . . . . . .  select an element from a matrix
+#O  MatElm( <mat>, <row>, <col> )  . . . . . .  select an entry from a matrix
+#O  <mat>[ <row>, <col> ]  . . . . . . . . . .  select an entry from a matrix
+##
+##  <#GAPDoc Label="MatObj_MatElm">
+##  <ManSection>
+##  <Oper Name="MatElm" Arg='mat, row, col'/>
+##
+##  <Returns>an entry of the matrix object</Returns>
+##
+##  <Description>
+##  For a matrix object <A>mat</A>, this operation returns the entry in
+##  row <A>row</A> and column <A>col</A>.
+##  <P/>
+##  Also the syntax <A>mat</A><C>[ </C><A>row</A><C>, </C><A>col</A><C> ]</C>
+##  is supported.
+##  <P/>
+##  Note that this is <E>not</E> equivalent to
+##  <A>mat</A><C>[ </C><A>row</A><C> ][ </C><A>col</A><C> ]</C>,
+##  which would first try to access <A>mat</A><C>[ </C><A>row</A><C> ]</C>,
+##  and this is in general not possible.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
 ##
-DeclareOperationKernel( "[,]",
-    [ IsMatrixObj, IS_INT, IS_INT ],
-    ELM_MAT );
-DeclareSynonym( "MatElm", ELM_MAT );
+DeclareOperationKernel( "MatElm", [ IsMatrixObj, IS_INT, IS_INT ], ELM_MAT );
+DeclareSynonym( "[,]", ELM_MAT );
```

Comment:
This change swaps the name of the kernel operation and its synonym. What is the motivation for this? (I don't see it mentioned in the commit message).

This changes e.g. backtraces that involve `[,]` (see the third commit in this PR, where I adjust `testspecial` for that, among other things). Also note that we for `[]`, we do this (and that was my main motivation for doing `DeclareOperationKernel("[,]", ...)`):
```gap
DeclareOperationKernel( "[]",
    [ IsList, IS_INT ],
    ELM_LIST );
```